src/main/java/com/appland/appmap/record/RecordingSessionFileStream.java

```diff
@@ -6,14 +6,20 @@
 
 import java.io.FileWriter;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
 
 /**
  * Receives recording data and writes it to disk.
  */
 public class RecordingSessionFileStream extends RecordingSessionGeneric {
```

Comment:
This class is going to be removed and all AppMaps will be streamed to temp files.

So, the long file name problem can only occur when the temp file is renamed to the "permanent" file name. There are two cases:

1) The file name is user-supplied - in which case the user should take care not to provide a too-long name
2) The file name is generated (e.g. from a test case). In this case, the generated name should honor the length restriction. So, this code is best moved to the hooks that save the AppMap after test case completion.